pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -856,6 +886,61 @@ func (r *ReconcileClusterDeployment) setupClusterInstallServiceAccount(namespace
 	return currentSA, nil
 }
 
+func (r *ReconcileClusterDeployment) selectorPodWatchHandler(a handler.MapObject) []reconcile.Request {
+	retval := []reconcile.Request{}
+
+	pod := a.Object.(*corev1.Pod)
+	if pod == nil {
+		// Wasn't a Pod, bail out. This should not happen.
+		log.Errorf("Error converting MapObject.Object to Pod. Value: %+v", a.Object)
+		return retval
+	}
+	if pod.Labels != nil {
+		if cdName, ok := pod.Labels[install.ClusterDeploymentNameLabel]; ok {
+			if cdNamespace, ok := pod.Labels[install.ClusterDeploymentNamespaceLabel]; ok {
+				retval = append(retval, reconcile.Request{NamespacedName: types.NamespacedName{
+					Name:      cdName,
+					Namespace: cdNamespace,
+				}})
+			}
+		}
+	}
+	return retval
+}
+
+func (r *ReconcileClusterDeployment) updateInstallRetriesTotalMetric(clusterDeploymentName string, clusterDeploymentNamespace string, cdLog log.FieldLogger) {
+	labels := map[string]string{"cluster_deployment": clusterDeploymentName, "namespace": "true"}
+	installerPodLabels := map[string]string{install.ClusterDeploymentNameLabel: clusterDeploymentName, install.ClusterDeploymentNamespaceLabel: clusterDeploymentNamespace, install.InstallJobLabel: "true"}
+	currentRetiesMetric, err := metricClusterDeploymentInstallRetriesTotal.GetMetricWith(labels)
+	if err != nil {
+		cdLog.Error("Error getting cluster deployment install retries total metrics.")
+	} else {
+		cdLog.Infof("CURRENT VALUE:%v", currentRetiesMetric)
+		pods := &corev1.PodList{}
+		err = r.Client.List(context.Background(), client.MatchingLabels(installerPodLabels), pods)
+		if err != nil {
+			log.WithError(err).Error("error listing pods")
+		} else {
+			podRestarts := 0
+			cdLog.Infof("pod length:%v", len(pods.Items))
+			for _, pod := range pods.Items {
+				cdLog.Infof("pod name:%v", pod.Name)
+				for _, cs := range pod.Status.ContainerStatuses {
+					if cs.Name == "installer" {
+						cdLog.Infof("Container Status:%v", cs.Name)
+						cdLog.Infof("POD RESTARTS:%v", cs.RestartCount)
+						podRestarts = int(cs.RestartCount)
+					}
+				}
+			}
+			//TODO - Calculates wrong metrics for now, need to figure out how to get metric value
```

Comment:
Uh oh, it looks like it may not be possible to read the current value of a counter, I suspect this is by design, something we shouldn't be doing.

Does this mean we should use a gauge so we can just set it to the value of the restarts on the pod? I know counters are preferred to gauges, and our value should only ever go up, but I can't see another way to work around this.